common/lib/capa/capa/capa_problem.py

```diff
@@ -978,7 +974,7 @@ def response_a11y_data(self, response, inputfields, responsetype_id, problem_dat
             for description in description_tags:
                 descriptions[
                     "description_%s_%i" % (responsetype_id, description_id)
-                ] = description.text
+                ] = HTML(stringify_children(description))
```

Comment:
@nasthagiri: At this point, I believe my question is answered.  

For data input from a course author, for the most part, here are the considerations:
1. Determine whether an input value allows HTML or only plain text.  Note that if we allow HTML, and a user enters plain text, that case would still be considered HTML input.
2. As in the `aria-label` example, if we allow HTML input and want to use it in a place that only allows plain text, we can use the bleach/strip method I described above.
3. Lastly, if we allow HTML input for a trusted user, we will by default use the full HTML as-is unless we have a clear reason (like #2) to strip out certain tags that would break the page if tags outside of some whitelist are used.  Again, we would use bleach to solve this issue.

As an aside, I've seen people use HTML to display icons next to the labels in the progress page, so it may not always be clear to us where people would like to enhance functionality through having HTML available.  During the safe scripting work, we have changed some places that seem like they would/should be text only to plain text (by escaping), and people have lost some of this flexibility.  So, it isn't always a simple call.